webapp/test/datastore_test.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:convert' show json;
+
+import 'package:test/test.dart';
+
+import 'package:CodaV2/datastore.dart' as ds;
+
+void main() {
+  test("Empty dataset", () {
+    ds.Dataset dataset = new ds.Dataset('Name');
+    expect(dataset.messages, []);
+    expect(dataset.codeSchemes, []);
+  });
+
+  test("Simple dataset from JSON", () {
+    Map datasetMap = json.decode(testJsonCollection);
+    ds.Dataset dataset = new ds.Dataset.fromJson(datasetMap);
+
+    expect(dataset.messages.length, 2);
+    expect(dataset.codeSchemes.length, 1);
+    expect(dataset.messages[0].messageID, "1");
+    expect(dataset.codeSchemes[0].schemeID, "scheme 1");
+  });
+}
+
+String testJsonCollection =
+"""{
+  "Name": "Test",
+  "Documents": [
+    {
+      "MessageID": "1",
```

Comment:
Please could you use a letter in this as well, so that if somewhere we are accidentally int parsing it, we'll fail

Reply:
Done, and also added letters in ValueID as well.